jmx/plugins/org.jboss.tools.jmx.jolokia/src/org/jboss/tools/jmx/jolokia/internal/connection/JolokiaMBeanServerConnection.java

```diff
@@ -246,39 +241,87 @@ public Object invoke(ObjectName name, String operationName, Object[] params, Str
+			return new J4pExecRequest(name, operationNameWithSignature);
+		} else {
+			return new J4pExecRequest(name, operationNameWithSignature, params);
+		}
+	}
 
-	
-	
-	/*
-	 * The following methods are *CURRENTLY* unsupported due to missing classname in json response. 
-	 */
-	
+	private String createOperationNameWithSignature(String operationName, String[] signature) {
+		StringJoiner stringJoiner = new StringJoiner(",", "(", ")");
+		Stream.of(signature).forEach(stringJoiner::add);
+		return operationName + stringJoiner.toString();
+	}
 
 	@Override
 	public ObjectInstance getObjectInstance(ObjectName name) throws InstanceNotFoundException, IOException {
-		// TODO Auto-generated method stub
-		return null;
+		return createObjectInstance(name);
 	}
 
 	@Override
-	public boolean isInstanceOf(ObjectName name, String className) throws InstanceNotFoundException, IOException {
-		// TODO Auto-generated method stub
-		return false;
-	}
-	@Override
 	public Set<ObjectInstance> queryMBeans(ObjectName name, QueryExp query) throws IOException {
-		// TODO Auto-generated method stub
-		return Collections.emptySet();
+		Set<ObjectInstance> res = new HashSet<>();
+		try {
+			J4pSearchRequest req = new J4pSearchRequest(name.getCanonicalName());
+			J4pResponse<J4pSearchRequest> j4pResponse = j4pClient.execute(req);
+			Object value = j4pResponse.getValue();
+			if(value instanceof JSONArray){
+				for (Object mbean : (JSONArray)value) {
+					if(mbean instanceof String){
+						res.add(createObjectInstance((String)mbean));
+					}
+				}
+			}
+		} catch (MalformedObjectNameException | J4pException e) {
+			Activator.pluginLog().logError(e);
+		}
+		return res;
+	}
+
+	private ObjectInstance createObjectInstance(String mbean) throws MalformedObjectNameException {
+		ObjectName objectName = new ObjectName(mbean);
+		return createObjectInstance(objectName);
+	}
+
+	private ObjectInstance createObjectInstance(ObjectName objectName) {
+		String classname = retrieveClassName(objectName);
+		return new ObjectInstance(objectName, classname);
+	}
+
+	private String retrieveClassName(ObjectName objectName) {
+		String escapedCanonicalPropertyList = objectName.getCanonicalKeyPropertyListString().replaceAll("/", "!/");
+		J4pListRequest listAttributes = new J4pListRequest(objectName.getDomain()+"/"+escapedCanonicalPropertyList+"/class");
+		try {
+			J4pResponse<J4pListRequest> listAttributesResponse = j4pClient.execute(listAttributes);
+			return listAttributesResponse.getValue();
+		} catch (J4pException e) {
+			Activator.pluginLog().logError(e);
+		}
+		return "";
 	}
 	
 	
+	@Override
+	public boolean isInstanceOf(ObjectName name, String className) throws InstanceNotFoundException, IOException {
+		String mBeanClass = retrieveClassName(name);
+		if(className != null && className.equals(mBeanClass)){
+			return true;
+		}
+		try {
+			return Class.forName(mBeanClass).isInstance(Class.forName(className));
+		} catch (ClassNotFoundException e) {
```

Comment:
I'm not sure if Class.forName() is the best way to accomplish this in an eclipse environment... but I really can't think of any alternative =/